### ISSUE TITLE
What are workaround steps if Windows user name contains spaces?

### ISSUE THREAD

**Issue by JackCAttack** [conda/conda#8725]
**The issue**
Various tools based on Conda (the Anaconda command line and Navigator in my case) still cannot handle spaces in Windows folder names.  I don't think that's in dispute.   So what are the workaround steps if your Windows user name (and thus your user-specific folders) has a space in it?  Whatever the workaround steps are, the Anaconda installation prompts need to clearly identify and suggest them.  In the meantime, I need to know for myself.

**My specs**
OS: Windows 7 Professional
Conda Version:  4.6.11, build py37_0
Anaconda download version:  Anaconda3 2019.03 64-bit
Spyder version: 3.3.4
Navigator Version:  1.9.7

**My experience**
I have one and only one user on my Windows 7 computer, which is "Jack Attack", and I cannot change that.  So every user-specific folder on my computer has a space in the name.  As a result I've had several problems getting Anaconda and Spyder to function properly.  It would be difficult to provide a dump of all my commands and output (and even more difficult for others to go through it) but suffice it to say that my issue isn't whether spaces are the problem but rather how to work around it.  I've uninstalled and reinstalled Anaconda several times and finally selected "C:\Anaconda" as the installation directory to avoid the space issue, as some have suggested.  But various operations within Anaconda continue to try to use "C:\Users\Jack Attack".  For example, it still insists on looking for the .condarc file in C:\Users\Jack Attack.  Anaconda creates a shortcut to the command line tool, and that shortcut tells the tool to open in the %home% directory, which on my computer is defined as C:\Users\Jack Attack.  

Uninstalling and reinstalling Anaconda doesn't fix the problem, possibly because some things are remembered from the prior install.  Maybe some references to C:\Users\Jack Attack are still in the registry too.

The install process and docs should identify each and every workaround step to get Anaconda to run on Windows accounts with spaces, since Windows folder names have had spaces in them for many years.

**Comment by JackCAttack** [comment]
I finally got Anaconda fully working by uninstalling and also manually removing anything that looked remotely related to Anaconda, including the directories .ipython, .anaconda and .conda under C:\Users\Jack Attack (which deletes existing environments, FYI).  Then I did a full Anaconda reinstall (again with C:\Anaconda as the install directory).  It STILL puts those three directories under C:\Users\Jack Attack, and puts the .conda file under C:\Users\Jack Attack too it seems.  And the default (home) directory when using Anaconda from the command line is still C:\Users\Jack Attack.  But everything now works.  

So my failed Anaconda operations might have failed for some other reason, not the spaces.  After my final reinstall, I did some other things differently too, like making sure Navigator was closed before using the command line interface.  But still, when I do certain tasks in the Anaconda command line, I get the warning that spaces may cause problems.  If spaces are NOT a problem, let's fix Anaconda so it doesn't display the warning, which can lead to wild goose chases and going down rabbit holes (after the geese?) or needlessly changing Windows userids or just giving up.  If spaces ARE a problem, well, that's a serious flaw in Anaconda too.

**Comment by jjhelmus** [comment]
We try hard to make conda and the Anaconda installer work when installed on Windows in paths with spaces.  That said, there are a number of software packages which can be installed using conda which do have issues with spaces.  These are open source packages which are not maintained by Anaconda, Inc.  Although we attempt to fix these packages when possible and submit the change back upstream, we cannot address all of these.  

For this reason we recommend that Anaconda be installed into a path without spaces.  Not because conda does not work with these paths but because various other third-party software that can be installed using conda may have issue.

**Comment by JackCAttack** [comment]
Thanks for your reply!  To Anaconda's credit, it is truly the only good tool for managing packages and environments.  But with great power comes great responsibility... The warning itself is harmful because it leads to time wasted and needless changing of user names (I even switched computers), so the Anaconda team should take the matter very seriously.  More thought needs to go into crafting a warning message that's more informative.  Perhaps it can say what you just said in your reply, with a link to an even more detailed explanation.  Or better yet, _remove_ the warning and penalize package owners who can't handle spaces, like refusing to install them or calling them out during installation.  If a package cannot handle spaces in 2019, it should be considered a package bug like any other, and it shouldn't be Anaconda's responsibility to display the warning.  As an analogy, I wouldn't expect Apple to warn me that some iPhone apps have bugs (and I believe Apple will kick you off the app store if it finds out).  Perhaps there should be a package certification process under which Anaconda would decertify buggy packages.  

If Anaconda Inc. feels it MUST display a warning, the warning should be precisely worded to say that a small number of third-party packages can't handle spaces.  It could also note how path problems are likely to manifest themselves - errors during package install? ...during package use?  And for the warning to recommend that Anaconda itself not be installed in a path without spaces is quite extreme, because it leads the user to change user names or create a new user as the first thing he tries when he has installation problems.  With a better warning message, the user might instead research which packages can't handle spaces.  Perhaps he'd post something to a forum asking "which of these packages can't handle spaces?" with the list of packages he's trying to install.  And if there were a package certification process, Anaconda could even note which packages aren't certified in its warnings during package installs.

That said, I certainly appreciate Anaconda and its great development team.

**Comment by mingwandroid** [comment]
The tools we speak of areas things like gnu make which is critical to building or packages. 

To gnu make this is not finding that can be fixed now. The horse has bolted. 

I agree our messaging could be better though.

**Comment by mingwandroid** [comment]
The problems usually present themselves at runtime.

**Comment by M1369** [comment]
Please @JackCAttack, how did you get to fix that issue because I am frustrated right now, trying to get anaconda to work for days now.

**Comment by JackCAttack** [comment]
> 
> 
> Please @JackCAttack, how did you get to fix that issue because I am frustrated right now, trying to get anaconda to work for days now.

As I said above, I reinstalled Anaconda after uninstalling it _thoroughly_ , and it worked.  So the Anaconda operations I was having trouble with might have failed for some other reason, not the spaces.  I wrote, "After my final reinstall, I did some other things differently too, like making sure Navigator was closed before using the command line interface."  

I'm not sure I can help you beyond that.  Perhaps try to create a username with no spaces and see if it works then.  Good luck!
